Gradle prod build: error: package org.h2.server.web does not exist

<!--
- Please follow the issue template below for bug reports.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**

<!-- Explain the bug, if an error is being thrown a stack trace helps -->
In JHI4 I used this command to build my gradle powered application using 

``` bash
./gradlew -P prod build buildDocker
``` 

which always did a great job, as it calls all required task, running tests and so on.

With JH5 this doesn't work anymore. It looks like h2 is compiled only for the default gradles dev profile, not when prod is enabled.

A workaround to this would be to 

```
./gradlew -x test -P prod build buildDocker
```

or using CI cache/artifacts to do it in several steps, like

```
./gradlew build
./gradlew test
./gradlew -P prod bootWar buildDocker
```

I think, that behavior should not be like this.

##### **Motivation for or Use Case**

<!-- Explain why this is a bug for you -->

I want to do an entire build, test, dockerization in one command for the most simple CI/dev setup.

##### **Reproduce the error**

<!-- For bug reports, an unambiguous set of steps to reproduce the error -->

Generate a (microservice) application with gradle chosen as build-tool. Run the above command and you will see something like this:

```
> Task :compileTestJava FAILED
/home/david/jhipster/samples/bookstore/bookservice/src/test/java/com/mycompany/myapp/config/WebConfigurerTest.java:12: error: package org.h2.server.web does n
ot exist
import org.h2.server.web.WebServlet;
                        ^
/home/david/jhipster/samples/bookstore/bookservice/src/test/java/com/mycompany/myapp/config/WebConfigurerTest.java:78: error: cannot find symbol
        verify(servletContext, never()).addServlet(eq("H2Console"), any(WebServlet.class));
                                                                        ^
  symbol:   class WebServlet
  location: class WebConfigurerTest
/home/david/jhipster/samples/bookstore/bookservice/src/test/java/com/mycompany/myapp/config/WebConfigurerTest.java:90: error: cannot find symbol
        verify(servletContext).addServlet(eq("H2Console"), any(WebServlet.class));
                                                               ^
  symbol:   class WebServlet
  location: class WebConfigurerTest
3 

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler error output for details.

````

##### **Suggest a Fix**

<!-- If you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->

The most straightforward one would be include be making h2 compile in general, and not just for dev profile. But this would add h2 to all prod builds as well. 

A better way would be to figure out, how to **not** add h2 in prod, except we are running compileJavaTest

##### **JHipster Version(s)**

JHipster 5.0.1

<!--
Which version of JHipster are you using, is it a regression?
-->

- [X] Checking this box is mandatory (this is just to show you read everything)

--- a/generators/server/templates/gradle/profile_prod.gradle.ejs
+++ b/generators/server/templates/gradle/profile_prod.gradle.ejs
@@ -23,7 +23,7 @@ apply plugin: 'com.moowork.node'
 <%_ } _%>
 
 dependencies {
-
+    testCompile "com.h2database:h2"
 }
 
 def profiles = 'prod'

